### PR TITLE
Validation in login and join modals 

### PR DESCRIPTION
**Pull request by HemaSAli**
#### Using validation in login and join modals 

Relates to #315

### CODE REVIEW

File: client/src/components/HomePage/modal/JoinModal/index.jsx [hunk 1]
```diff
@@ -17,8 +18,22 @@ class Join extends Component {
         firstName, lastName, email, password,
       } = context.data;
       if (firstName && lastName && email && password) {
-        this.setState({ signingUp: false });
-        context.updateState({ completeJoin: true });
+        if (!validator.isEmail(email)) {
+          this.setState({ signingUp: false });
```
Comment (tomduggan85) [inline client/src/components/HomePage/modal/JoinModal/index.jsx]: @HemaSAli 
It looks like there a few common lines of code that could be refactored into a function, to reduce some duplication:  The setState which sets signingUp to false and the setPopupMessage could turn into something like `this.showSignupError('please enter a valid email address!')`

Comment (HemaSAli) [reply client/src/components/HomePage/modal/JoinModal/index.jsx]: @tomduggan85 
Thanks Tom, you can see now..
I have created a function which is like this 
```
  setPopUp = (message, title, context) => {
    context.setPopUpMessage({
      message, title,
    });
  }
```

Comment (HemaSAli) [reply client/src/components/HomePage/modal/JoinModal/index.jsx]: also I improved it to be like this
```
  setPopUp = (message, title, context, signUp) => {
    context.setPopUpMessage({
      message, title,
    });
      this.setState({ signingUp: false });
  }
```
So I can send setState in it

File: client/src/components/HomePage/modal/JoinModal/index.jsx [hunk 2]
```diff
@@ -10,21 +11,31 @@ import LoadingModal from '../LoadingModal';
...
     setTimeout(() => {
       const {
         firstName, lastName, email, password,
       } = context.data;
       if (firstName && lastName && email && password) {
-        this.setState({ signingUp: false });
-        context.updateState({ completeJoin: true });
+        if (!validator.isEmail(email)) {
+          this.setPopUp('please Enter valid email adress !', 'Error !', context);
+        } else if (!validator.isAlpha(firstName) || !validator.isAlpha(lastName)) {
```
Comment (FarahZaqout) [inline client/src/components/HomePage/modal/JoinModal/index.jsx]: some names might include symbols, like O'neal and other names.

File: client/src/components/HomePage/modal/JoinModal/index.jsx (hunk 1 above)
Comment (tomduggan85) [reply client/src/components/HomePage/modal/JoinModal/index.jsx]: Cool, thanks!

**Pull request merged**